vignettes/r-developer-guide.Rmd

```diff
@@ -24,129 +24,178 @@ knitr::read_chunk(path = system.file("plumber", "loess", "plumber.R", package =
 y <- 1/x^2 + rnorm(length(x), sd = 0.05)
 fit <- loess(y ~ x, span = 0.75)
 y_fit <- predict(fit, data.frame(x, y))
-```
 
-Here, we can visualize the result of our smoothed fit:
-```{r, fig.width=8, fig.height=6}
+# Plot the data and prediction.
 plot(x, y)
 lines(x, y_fit)
 ```
 
-To turn this into a Tableau extension, we can use `plumber` annotations to define the endpoint:
+## The Anatomy of a plumbertableau Extension {#anatomy}
+
+Here's that same model in a finished plumbertableau extension.
+
 ```{r loess, eval = FALSE}
 ```
 
-### Development {#development}
-Now that we have the whole picture, let's break down each piece.
+There are a few parts to our extension which are required to make it work:
 
-#### Plumber annotations
-```{r, eval = FALSE}
-library(plumber)
-library(plumbertableau)
+- **Annotations**. Comments beginning with `#*` are [annotations](https://www.rplumber.io/articles/annotations.html), which define aspects of the API for Plumber and plumbertableau.
+- **Function definitions** — Each function definition must use annotations to describe the data types for all inputs and the value it returns.
+- **The Tableau extension footer** — This contains the `tableau_extension` object with the `@plumber` annotation. This does the work of modifying the Plumber router to be Tableau-compatible.
 
-#* @apiTitle Loess Smoothing
-#* @apiDescription Loess smoothing for Tableau
+The `@apiTitle` and `@apiDescription` annotations are used to describe the extension in documentation which is generated.
 
-#* Fit a loess curve to the inputs and return the curve values
-#* @param alpha Degree of smoothing
-#* @tab.arg x:integer X values for fitting
-#* @tab.arg y:numeric Y values for fitting
-#* @tab.return numeric Fitted loess values
-#* @post /predict
-```
+### Annotating Function Definitions for plumbertableau
 
-Here we load the relevant packages and use standard `plumber` annotations to describe the API. `plumbertableau` introduces new annotations `tab.arg` and `tab.return` that allow you to describe the expected Tableau arguments and types and the return value and type of the API.
+Let's look again at our un-annotated function definition.
 
-#### Function definition
-```{r, eeval = FALSE}
+```{r, eval = FALSE}
+#* Annotate me!
 function(x, y, alpha = 0.75) {
   alpha <- as.numeric(alpha)
   l_out <- loess(y ~ x, span = alpha)
   predict(l_out, data.frame(x, y))
 }
 ```
 
-This is the actual function definition for the endpoint. Here, the values `x` and `y` are the values described in the `tab.arg` statements. These values are automatically parsed and named from the request submitted from Tableau. `alpha` is a URL parameter we can use to adjust the level of smoothing.
+It accepts three inputs: `x`, `y`, and `alpha`. It fits a `loess()` model using this data, and returns the output of the `predict()` function for that model. We need to annotate its inputs and output so that plumbertableau can translate the data correctly for Tableau.
```

Comment:
```suggestion
It accepts three inputs: `x`, `y`, and `alpha`, fits a `loess()` model using this data, and returns the output of the `predict()` function for that model. We need to annotate the function inputs and output so that plumbertableau can translate the data correctly for Tableau and generate proper documentation for the extension.
```